docs/privacy/cspp/overview.md

```diff
@@ -36,35 +36,47 @@ The team ended up pivoting to CoinShuffle++, a simpler and more DoS-resistant so
+The original CoinShuffle++ protocol uses a P2P approach to coin mixing, while it requires a server to handle communication, it does not expect it to do any computation. 
 
-This enables the use of a **client-server** architecture in Decred's first CSPP implementation. It is a common approach to networks and it has two benefits: **enhanced privacy and simplicity**.
+CSPP's threat model, the variety of attacks it takes into account, includes the scenario where the attacker owns the server, hosts the service and handles communication. CSPP ensures anonymity even while trying to join a session hosted by a malicious server. 
 
-Having a P2P network would mean **every peer could see other's public IP**, which would compromise privacy. Users could **run their own servers**, but they would have **limited utility** because the goal is to get as many peers as possible in each mix.
+This enables the use of a **client-server** architecture. It is a common approach to networks and it has three benefits: enhanced privacy, simplicity and mixing times.
+
+Decred's first implementation uses a centralized server as a coordination point and a optimized polynomial factorization solver (to improve mixing times). It enhances simplicity by having a single authority responsible for timeouts and advancing the protocol states.  Having a single outbound connection also means peers do not rely on NAT or UDP hole-punching to communicate with each other.
+
+Depending on a P2P network would mean every peer could see other's public IP, which would compromise privacy. Users could run their own servers, but they would have limited utility because the goal is to get as many peers as possible in each mix.
+
+### Handling malicious peers
+
+The protocol handles collisions and malicious peers using a privacy-preserving redundant messaging step. In its worst case scenario, it requires `4+2f` communication rounds for `f` malicious peers. To achieve this, it that each peer uses a freshly generated bitstring (verification key).
+
+After a failed protocol run, peers can reveal session secrets to expose and exclude malicious disruptors, and move on with the mix. If a dishonest peer is excluded from the mix, all secrets so far will be discarded and new ones are generated.
 
 ---
 
 ## Benefits
 
-The implementation of the CSPP mixnet did not require changing the consensus rules. It follows Decred's low-hanging fruit approach and it allowed Decred developers to work on it in a permisionless manner.
+The implementation of the CSPP mixnet did not require changing the consensus rules. It is an opt-in implementation that  obfuscates ownership of DCR coins with out requiring any modifications to the consensus rules. A non-opt-in system would be much more complex.
```

Comment:
> A non-opt-in system would be much more complex."

I think this extra sentence should be removed unless some extra justification is added. Discussing the pro and cons of opt-in/opt-out is a whole new can of worms.